django/core/management/commands/shell.py

```diff
@@ -101,11 +101,16 @@ def handle(self, **options):
             exec(sys.stdin.read())
             return
 
-        available_shells = [options['interface']] if options['interface'] else self.shells
-
-        for shell in available_shells:
+        if options['interface']:
+            shell = options['interface']
             try:
                 return getattr(self, shell)(options)
             except ImportError:
-                pass
-        raise ImportError("Couldn't load any of the specified interfaces.")
```

Comment:
What about keeping the old logic and just changing the message to `raise CommandError("Couldn't import the %s interface." % shell)`. The error should only be trigger when specifying a specific shell anyway.